Sources/TuistDependencies/DependenciesManagerInteractors/CarthageInteractor.swift

```diff
@@ -0,0 +1,139 @@
+import RxBlocking
+import TSCBasic
+import TuistCore
+import TuistSupport
+
+// MARK: - Carthage Interactor Errors
+
+enum CarthageInteractorError: FatalError, Equatable {
+    /// Thrown when CocoaPods cannot be found.
+    case carthageNotFound
+
+    /// Error type.
+    var type: ErrorType {
+        switch self {
+        case .carthageNotFound:
+            return .abort
+        }
+    }
+
+    /// Error description.
+    var description: String {
+        switch self {
+        case .carthageNotFound:
+            return "Carthage was not found either in Bundler nor in the environment"
+        }
+    }
+}
+
+// MARK: - Carthage Interacting
+
+public protocol CarthageInteracting {
+    /// Installes `Carthage` dependencies.
+    /// - Parameter path: Directory whose project's dependencies will be installed.
+    /// - Parameter method: Installation method.
+    /// - Parameter dependencies: List of dependencies to intall using `Carthage`.
+    func install(
+        at path: AbsolutePath,
+        method: InstallDependenciesMethod,
+        dependencies: [CarthageDependency]
+    ) throws
+}
+
+// MARK: - Carthage Interactor
+
+public final class CarthageInteractor: CarthageInteracting {
+    private let fileHandler: FileHandling!
+    private let cartfileResolvedInteractor: CartfileResolvedInteracting
+    private let carthageFrameworksInteractor: CarthageFrameworksInteracting
+
+    public init(
+        fileHandler: FileHandling = FileHandler.shared,
+        cartfileResolvedInteractor: CartfileResolvedInteracting = CartfileResolvedInteractor(),
+        carthageFrameworksInteractor: CarthageFrameworksInteracting = CarthageFrameworksInteractor()
+    ) {
+        self.fileHandler = fileHandler
+        self.cartfileResolvedInteractor = cartfileResolvedInteractor
+        self.carthageFrameworksInteractor = carthageFrameworksInteractor
+    }
+
+    #warning("TODO: The hardes part here will be knowing whether we need to recompile the frameworks")
+    public func install(
+        at path: AbsolutePath,
+        method: InstallDependenciesMethod,
+        dependencies: [CarthageDependency]
+    ) throws {
+        // determine platforms
+        let platoforms: Set<Platform> = dependencies
+            .reduce(Set<Platform>()) { platforms, dependency in platforms.union(dependency.platforms) }
+
+        try fileHandler.inTemporaryDirectory { temporaryDirectoryPath in
```

Comment:
Unless I've missed it, I don't see logic to remove any existing frameworks that are no longer needed. Let's say we have the following scenario:
- Frameworks A, B, and C exist locally
- You build frameworks 1, 2, and C.

I'd expect 1 and 2 to be removed, A and B to show up in the directory, and C to only be built if it's necessary. I've seen that doing the latter is part of a future improvement, but what about the selection of 1 and 2?

Here's my suggestion on how to implement it:
1. Get the list of existing frameworks.
2. Get the list of built frameworks after running the Carthage command.
3. Diff them and determine which ones should be deleted and which ones copied.

Reply:
It is more tricky. What if frameworks from `Diff them and determine which ones should be deleted` have been installed via `Cocoapods` or `SPM`? I should delete them just because they don't exist in `Get the list of built frameworks after running the Carthage command.`

Right now Im going to provide simply diffing logic (like you proposed) assuming that we have only `carthage` but it will need to be improved in the future.

Reply:
- I refactored `CarthageFrameworksInteractor`
- I introduced `Tuist/Directory/graph.json`
- I added `CarthageFrameworksInteractorTests.test_save_remove_existing`

For sure diffing will need improvements. I think it will be moved outside `CarthageInteractor` for better diffing dependencies between different dependencies managers. 

In this PR Im introducing "initial" solution.